src/main/java/com/openlattice/chronicle/services/enrollment/EnrollmentService.java

```diff
@@ -86,6 +88,29 @@ private void updateDeviceIdsCache( UUID organizationId, UUID deviceEKID, String
         scheduledTasksManager.getDeviceIdsByEKID().put( datasourceId, deviceEKID );
     }
 
+    private Map<UUID, Set<Object>> createDeviceData( String datasourceId, Optional<Datasource> datasource ) {
+        Map<UUID, Set<Object>> data = Maps.newHashMap();
+        data.put( edmCacheManager.getPropertyTypeId( STRING_ID_FQN ), ImmutableSet.of( datasourceId ) );
+
+        if ( datasource.isPresent() ) {
+            if ( AndroidDevice.class.isAssignableFrom( datasource.get().getClass() ) ) {
+                AndroidDevice device = (AndroidDevice) datasource.get();
+                data.put( edmCacheManager.getPropertyTypeId( MODEL_FQN ), ImmutableSet.of( device.getModel() ) );
+                data.put( edmCacheManager.getPropertyTypeId( VERSION_FQN ), ImmutableSet.of( device.getOsVersion() ) );
+            }
+
+            if ( IOSDevice.class.isAssignableFrom( datasource.get().getClass() ) ) {
+                IOSDevice device = (IOSDevice) datasource.get();
+                data.put( edmCacheManager.getPropertyTypeId( MODEL_FQN ),
+                        ImmutableSet.of( device.getLocalizedModel() ) );
+                data.put( edmCacheManager.getPropertyTypeId( VERSION_FQN ), ImmutableSet.of( device.getVersion() ) );
+                data.put( edmCacheManager.getPropertyTypeId( NAME_FQN ), ImmutableSet.of( device.getName() ) );
```

Comment:
Why don't android devices utilize name?

Reply:
AndroidDevice
(https://github.com/openlattice/chronicle-api/blob/main/src/main/java/com/openlattice/chronicle/sources/AndroidDevice.java) is constructed using a subset of these static properties (https://developer.android.com/reference/android/os/Build) which does not include device name. iOS (https://developer.apple.com/documentation/uikit/uidevice) surfaces name